src/pybind/mgr/dashboard/tools.py

```diff
@@ -764,3 +766,31 @@ def str_to_bool(val):
     if isinstance(val, bool):
         return val
     return bool(strtobool(val))
+
+
+def restart_if_running():
```

Comment:
It is probably better to send a "mgr fail" command to the monitor.  That way the monitor knows that this is something intentional, and can avoid logging a warning about an unexpected daemon restart.

(ceph-mgr daemons notice if they were killed, and respawn themselves)